brownie/network/multicall.py

```diff
@@ -0,0 +1,58 @@
+from contextlib import contextmanager
+from dataclasses import dataclass
+from functools import partial
+from types import FunctionType
+from typing import List
+
+from wrapt import ObjectProxy
+
+from brownie import Contract
+
+
+@dataclass
+class Call:
+    call: List
+    decode_output: FunctionType
+
+
+class Caller:
+    def __init__(self, queue, contract=None, func=None):
+        self.queue = queue
+        self.contract = contract
+        self.func = func
+
+    def __getattr__(self, name):
+        return Caller(self.queue, self.contract, getattr(self.contract, name))
+
+    def __call__(self, *args, **kwds):
+        future = Result(
+            Call(
+                [str(self.contract), self.func.encode_input(*args, **kwds)],
+                self.func.decode_output,
+            )
+        )
+        self.queue.append(future)
+        return future
+
+
+class Result(ObjectProxy):
+    def __repr__(self):
+        return repr(self.__wrapped__)
+
+
+@contextmanager
+def multicall(block_identifier=None):
+    """
+    with multicall() as caller:
+        response = caller(contract).func(args)
+    """
+    queue = []
+    yield partial(Caller, queue)
+
+    multicall2 = Contract("0x5BA1e12693Dc8F9c48aAD8770482f4739bEeD696")
```

Comment:
Suggestion: make this a default kwarg for multicall so it's not hard-coded in the framework?

```
@contextmanager
def multicall(block_identifier=None, multicall2_address="0x5BA1e12693Dc8F9c48aAD8770482f4739bEeD696"):
    ...
    multicall2 = Contract(multicall2_address)
```

Reply:
I was thinking to make it even more flexible:
- live network => use `eth_call` code override to make it available pre-deploy
- dev network => seamlessly deploy a contract and cache it

Reply:
I think `eth_call` code override should be optional as some providers don't support them (e.g. infura)